server/controllers/message.js

```diff
@@ -0,0 +1,99 @@
+const express = require("express");
+const router = express.Router();
+const mongoose = require("mongoose");
+mongoose.Promise = global.Promise;
+const Convo = require("../models/convo");
+const Message = require("../models/message");
+const ObjectId = require("mongodb").ObjectId;
+
+// POST -- Add Message
+module.exports.addMessage = async (req, res, next) => {
+    const convoId = req.params.convoId;
+
+    const newMessage = new Message(req.body);
+
+    let convoUsers;
+
+    // Gets users involved in conversation
+    Convo.findById(convoId, (err, convo) => {
+        if (err) {
+            console.log(err);
+            res.send(err);
+        } else {
+            convoUsers = convo.users;
+        }
+    }).exec((err) => {
+        if (err) {
+            console.log(err);
+            res.send(err);
+        } else {
+            // Ensures author is a member of the conversation
+            if (convoUsers.includes(newMessage.author)) {
+                Convo.updateOne(
+                    { _id: convoId },
+                    { $push: { messages: newMessage } },
+                    (err) => {
+                        if (err) {
+                            console.log(err);
+                            res.send(err);
+                        } else {
+                            res.json({ success: true, msg: "Message added" });
+                        }
+                    }
+                );
+            } else {
+                res.status(400);
+                res.send("Message sender not a conversation user!");
+            }
+        }
+    });
+};
+
+// PUT -- Edit Message
+module.exports.editMessage = async (req, res, next) => {
+    const convoId = req.params.convoId;
```

Comment:
```suggestion
    const {convoId, messageId} = req.params.convoId;
```

to reduce repeating code and improve on readability